backend/open_webui/utils/oauth.py

```
@@ -238,206 +240,336 @@ async def login(self, request, provider):
+        
+        try:
+            if provider_name in OAUTH_PROVIDERS:
+                return OAUTH_PROVIDERS[provider_name]
+            log.error(f"Provider {provider_name} not found in OAUTH_PROVIDERS")
+            return None
+        except Exception as e:
+            log.error(f"Error getting provider data for {provider_name}: {str(e)}")
+            log.exception(e)
+            return None
+    
+    async def _get_token(self, request, provider_name):
```

Ruff (0.8.2)

Missing return type annotation for private function `_get_token`

(ANN202)